tools/lxdclient/client.go

```diff
@@ -52,11 +56,7 @@ var lxdLoadConfig = lxd.LoadConfig
 func newRawClient(remote, configDir string) (*lxd.Client, error) {
 	logger.Debugf("loading LXD client config from %q", configDir)
 
-	// This will go away once LoadConfig takes a dirname argument.
-	origDirname := updateLXDVars(configDir)
-	defer updateLXDVars(origDirname)
-
-	cfg, err := lxdLoadConfig()
+	cfg, err := lxdLoadConfig(path.Join(configDir, "config.yml"))
```

Comment:
On Tue, Jan 26, 2016 at 08:27:30PM -0800, John Arbash Meinel wrote:
> > @@ -52,11 +56,7 @@ var lxdLoadConfig = lxd.LoadConfig
> >  func newRawClient(remote, configDir string) (*lxd.Client, error) {
> >  	logger.Debugf("loading LXD client config from %q", configDir)
> >  
> > -	// This will go away once LoadConfig takes a dirname argument.
> > -	origDirname := updateLXDVars(configDir)
> > -	defer updateLXDVars(origDirname)
> > -
> > -	cfg, err := lxdLoadConfig()
> > +	cfg, err := lxdLoadConfig(path.Join(configDir, "config.yml"))
> 
> So we do have the issue that we don't generally let Providers (and thus Broker/container.Managers) have direct DB access. So we'd need to be passing this information in via just extra parameters to Init or Config.
> What are the things we need, certificates and the config that says to use "local"?

We don't need any certificates if you only want to talk to the LXD
provider on the local host, just enough perms to write to
/var/lib/lxd/unix.socket

> I know we have support for some of those things at the Provider level, to date we haven't needed them for Brokers. Will need some thought as you need a custom cert for every machine to every LXD, right? How do we validate the LXD's certificate, or do we just accept it as long as it is valid?

If you're only talking to the local lxd, since it's the only thing
that can listen on /var/lib/lxd/unix.socket, so there is no
authentication required. Things get a little more complicated when you
want to talk to remotes, but as far as I can tell we don't need to do
that (at least right now with the LXD provider/container type).